frontend/hooks/player.ts

```diff
@@ -0,0 +1,33 @@
+import UIfx from "uifx";
+import {
+    useState,
+    useRef,
+    useEffect,
+    Dispatch,
+    SetStateAction,
+    MutableRefObject,
+} from "react";
+
+export function usePlayer(
+    audio: string
+): [
+    boolean,
+    Dispatch<SetStateAction<Boolean>>,
+    MutableRefObject<(() => void) | undefined>
+] {
+    const player = useRef<UIfx>();
+    useEffect(() => {
```

Comment:
This should have an empty dependency array so you don't create a new UIfx every time